pkg/kv/txn_interceptor_pipeliner.go

```diff
@@ -179,12 +179,14 @@ type txnPipeliner struct {
 	// to have succeeded. They will need to be proven before the transaction
 	// can commit.
 	ifWrites inFlightWriteSet
-	// The transaction's write footprint contains spans where intent writes have
-	// been performed at some point by the transaction. The span set contains
-	// spans encompassing all writes that have already been proven in this epoch
-	// and all writes, in-flight or not, at the end of prior epochs. All of the
-	// transaction's in-flight writes are morally in this set as well, but they
-	// are not stored here to avoid duplication.
+	// The transaction's lock footprint contains spans where locks have been
```

Comment:
Ah, I missed earlier that the pipeliner keeps track of the locks solely for the purpose of handing them to EndTransaction in the end. That makes sense, though it would be nice to clarify where the locks matter (include in txn record) and where the writes do (pipeline).